runelite-client/src/main/java/net/runelite/client/plugins/hiscore/NameAutocompleter.java

```diff
@@ -241,6 +260,27 @@ private boolean findAutocompleteName(String nameStart)
 		return autocompleteName.isPresent();
 	}
 
+	private void populateSearchHistory()
+	{
+		searchHistory = EvictingQueue.create(MAX_SEARCH_HISTORY);
+		Matcher m = RSN_PATTERN.matcher(config.searchHistory());
+
+		while (m.find())
+		{
+			searchHistory.add(m.group());
+		}
+	}
+
+	void addToSearchHistory(@NonNull String name)
+	{
+		if (!searchHistory.contains(name))
+		{
+			searchHistory.offer(name);
+			String result = searchHistory.stream().collect(Collectors.joining(","));
```

Comment:
You should be able to use `String::join`

Reply:
I used `Text::toCSV` here